src/rl_framework/util/saving_and_loading/connector/clearml_connector.py

```diff
@@ -83,19 +102,25 @@ def upload(
 
         assert file_name
 
-        # Step 1: Save agent to temporary path and upload .zip file to ClearML
-        with tempfile.TemporaryDirectory() as temp_path:
-            logging.debug(f"Saving agent to .zip file at {temp_path} and uploading artifact ...")
-
-            file_literal, file_ending = str.split(file_name, ".")
-            checkpoint_suffix = f"-{checkpoint_id}" if checkpoint_id else ""
-
-            agent_save_path = Path(os.path.join(temp_path, f"{file_literal}{checkpoint_suffix}.{file_ending}"))
-            agent.save_to_file(agent_save_path)
-            while not os.path.exists(agent_save_path):
-                time.sleep(1)
-
-            self.task.update_output_model(name=f"{file_literal}{checkpoint_suffix}", model_path=temp_path)
+        # Step 1: Save agent to temporary file and upload to ClearML
+        file_literal, file_ending = str.split(file_name, ".")
+        checkpoint_suffix = f"-{checkpoint_id}" if checkpoint_id else ""
+        file_name = f"{file_literal}{checkpoint_suffix}"
+
+        # Save agent to file
+        agent_save_path = Path(f"{str(uuid.uuid1())}-{file_name}.{file_ending}")
```

Comment:
Because ClearML is async, and once done deletes the file (by default). Thus as long as the process isn't killed, that path is actually "temporary". But you are right, it should still use the systems temp dir, will fix.

The reason the old solution stopped working is because the temp directory, and all its content, is removed when using the context manager (with tempfile...), before ClearML can upload. Yes, ClearML could upload in sync, but its not recommened. It worked previously because ClearML zipped the folder and used their own temp path.